scipy/stats/_circstats.py

```diff
@@ -0,0 +1,726 @@
+          minimize the circular mean deviation. Instead, the number of observations
+          within 90 degrees of a bisecting median is greater than the number of
+          observations within 90 degrees of its antipode.
+
+        See [3]_ for precise mathematical definitions.
+
+    high : float, optional
+        Upper boundary of the principal value of an angle.  Default is ``2*pi``.
+    low : float, optional
+        Lower boundary of the principal value of an angle.  Default is ``0``.
+    axis : int, default: 0
+        Axis along which the circular median is computed.
+
+    Returns
+    -------
+    circmedian : float
+        Circular median, restricted to the range ``[low, high]``.
+
+    See Also
+    --------
+    circmean : Circular mean.
+    circstd : Circular standard deviation.
+    circvar : Circular variance.
+
+    Notes
+    -----
+    There are several definitions of "circular median" in the literature. Seminal
+    references for circular statistics [1]_ and [2]_ mention the bisecting property
+    of the circular median and its connection with minimization of the circular mean
+    deviation. However, [1]_ prioritizes minimization of the circular mean deviation
+    as the defining property, whereas [2]_ requires that "the majority of the data
+    points are nearer to [the median] than [its antipode]".
+
+    Although the two definitions produce the same circular median in many cases
+    (unimodal distribution, no ties), [3]_ demonstrates that the two definitions can
+    lead to different - and in fact, entirely opposite - results. [3]_ also compares the
+    arc-distance and bisecting medians with two other definitions adopted from
+    vector-space literature, and it concludes that the arc-distance median is the only
+    one that provides all four properties under consideration. This, and the fact that
+    the arc-distance definition is used by default in other circular statistics software
+    (e.g. [4]_, [5]_, [6]_), contributed to the choice of ``'arc-distance'`` as the
+    default convention.
+
+    [7]_ addresses the fact that many points - even continuous arcs of the unit circle -
+    may satisfy the definition of a circular median. It proposes that "the estimate of
+    the population circular median be the average (circular mean) of all angles
+    satisfying the definition of median", and that "for odd samples, the candidate
+    values are the observations themselves". However, it suggests that "for even
+    samples, the candidate values are the midpoints of all neighboring observation".
+    `circmedian` always finds all points *among the observations* (and their antipodes,
+    in the ``'bisecting'`` case) that satisfy the chosen definition of a circular median
+    and returns their circular mean. If the circular mean is poorly defined (i.e. the
+    circular variance is with a small tolerance of ``1.0``), `circmedian` returns NaN.
```

Comment:
If I read correctly, four cases are discussed here. Take the 'arc-distance'. Let $A$ denote the set of all angles that minimise the total arc distance, $S$ the sample points that minimise the total arc distance, $M$ the "middle" of adjacent pairs of points in $S$. Then:

1. circular median = circular mean of $A$;
2. circular median $\in S$ if $|S|$ is odd;
3. circular median $\in M$ if $|S|$ is even;
4. circular median = circular mean of $S$.

To me (1) seems the "nicest", but I don't know how to find $A$ to compute (1). (4) seems the most practical. It might be the case that (1) and (4) are actually equivalent, but I cannot prove it nor provide a counterexample. I also wonder: Does (1) + $|S|$ odd imply (2)? Does (1) and $|S|$ even imply (3)?